modules/graceful-shutdown.adoc

```diff
@@ -77,3 +77,8 @@ Cluster administrators are responsible for ensuring a clean restart of their own
 ====
 
 . Shut off any cluster dependencies that are no longer needed, such as external storage or an LDAP server. Be sure to consult your vendor's documentation before doing so.
++
+[NOTE]
+====
+You cannot suspend the cloud resources in a cluster except for VMs.
```

Comment:
I read the assembly and the BZ, and I'm still a bit confused about what we're telling users to do. And what we're telling them not to do. You seem to be saying: When you're shutting off "cluster dependencies" it's okay to shut off "VMs" but not any other "cloud resource".

Does that mean that we can't shut off external storage or LDAP servers if they are cloud-based?

Is there any way to add detail to make this clearer?

Reply:
Thank you for your comment. The original question from a user that they want to remove loadbalancer to reduce the cost. However, it is not possible. To clarify our document, I think we need to have a note that describes users can shutdown the resources which are able to shutdown by user. Initially, I think the resources can be shut down are VMs. To make it precise, we need to describe it in details.

Reply:
Hi all! As far as I can see, the assembly that this module is included in relates to all types of OpenShift Container Platform installations and methods. This includes installations on cloud provider platforms, virtual machine infrastructure, bare metal, OpenStack Platform, etc. In context to that, this note seems too specific to the cloud provider deployment scenarios.

In addition to this, in my view there are some cloud provider resources in some deployment scenarios that are not virtual machines. For instance, if you deploy a cluster on AWS using CloudFormation, there are [resources in the AWS stack](https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-user-infra.html#installation-aws-user-infra-requirements_installing-aws-user-infra) that could be deleted after the cluster is removed.

From my understanding of the description in [BZ 2042709,](https://bugzilla.redhat.com/show_bug.cgi?id=2042709), the requirement is to include steps or a note on shutting down or deleting resources that a cloud-provider based cluster depends on, to prevent ongoing charges for resources that are not being used. To accommodate this in the short-term, you could update the note to be an important admonition and change it to something like the following:

> **Important**: If you deployed your cluster on a cloud-provider platform and you have deleted your cluster, you can remove the cloud resources that were used by your cluster. The resources might be used by other clusters in the same cloud account. You must only remove the resources if they are not required by other clusters. You must also have the required permissions in your cloud provider account to delete the resources.

If there is a requirement in the longer-term to provide detailed deletion steps for the cloud resources that a cluster depends on, then in my view this would probably involve the creation of cluster deletion assemblies that are specific to platforms deployment types and installation methods.

Reply:
@pneedle-rh , Thank you for your comment!
> From my understanding of the description in [BZ 2042709](https://bugzilla.redhat.com/show_bug.cgi?id=2042709), the requirement is to include steps or a note on shutting down or deleting resources that a cloud-provider based cluster depends on, to prevent ongoing charges for resources that are not being used. To accommodate this in the short-term, you could update the note to be an important admonition and change it to something like the following:

If I understand correctly, the original starting point for this BZ is that when the cu temporarily shut down a cluster to save costs, they also want to stop the ELB.
But regardless of OCP's design, the ELB cannot be stopped. The only way is to delete the ELB, in which case OCP will not be able to restore it properly.
Therefore, they request that this fact be clearly stated in the document.
(Yes, This issue may be AWS-specific...)

Reply:
@kazuhisya thank you for clarifying that. It sounds then like there is a need for a separate note, in addition to the one that I suggested, that relates explicitly to the ELB scenario.

Reply:
Hi @pneedle-rh ,
Thanks and agreed. I expect the following like sentences:  `if you delete cloud resources (includes Load balancer, authentication roles, etc.) required for OCP while the VM is suspended, OCP may not be restored properly.`

Reply:
@kazuhisya thank you for your suggestion.

@darshan-nagaraj please let @kazuhisya and I know if you need any further information in order to proceed. Thanks!